components/typography/styles.js

```diff
@@ -2,7 +2,7 @@ import '../colors/colors.js';
 import { css, unsafeCSS } from 'lit';
 
 export const _isValidCssSelector = (selector) => {
-	const re = /([a-zA-Z0-9-_ >.#]+)(\[[a-zA-Z0-9-_]+\])?([a-zA-Z0-9-_ >.#]+)?/g;
+	const re = /([a-zA-Z0-9-_ >.#:]+)(\[[a-zA-Z0-9-_]+\])?([a-zA-Z0-9-_ >.#]+)?/g;
```

Comment:
Interesting... it seems like we added a test that would explicitly _not_ allow `:host` as a selector, which is what I want. 🤔

Reply:
Looks like this was [added here](https://github.com/BrightspaceUI/core/pull/2831/files#r967259003) based on a suggestion from @dbatiste. Dave, do you remember why we didn't want to allow `:host` here? Just that we didn't need it at the time, or is there some danger I'm not considering?

Reply:
Yeah this is both pretty limiting and also a bit hamhanded, but looks like James was aware when it went in. It can be a separate item, but I'd like to get some more context on this and rewrite it.

Reply:
I also went to use this before my vacation and was similarly hamstrung.

But the reason was [this case](https://github.com/BrightspaceUI/core/pull/2698#discussion_r931527209). That is...

Given this:
```css
:host([skeleton]) ${selector}.d2l-skeletize::before
```

This would be no good if the consumer tried to do:
```javascript
generateLabelStyles(':host .some-element')
```

Reply:
I'd be ok with relaxing the validation, knowing that this is purely for internal use by us, and that there would still be cases that wouldn't work. Or maybe a different helper. 🤷‍♂️

Reply:
Aha, thanks for finding that. Ok... so maybe for now we treat `:host` completely separately, meaning that it could only be used on its own? And then if Danny wants to rewrite this later, that's cool. 😎

Reply:
Sounds good to me!